### DISCUSSION TITLE
Import problem

### DISCUSSION THREAD

**Discussion by ishanuda** [krisk/Fuse#605] [Q&A] [answered]
Hi,
Wish you a very happy new year for all the community members.

I am trying to import fuse.js in to the Node.JS application as follows.

`import Fuse from 'fuse.js';`

By importing like the above I cannot construct a new object or create a new index and it gives the following errors.

1. Create the fuse obj:
`const fuse = new Fuse(data, options);`

Error:
fuse_js_1.default is not a constructor

2. Create the index:
`const index = Fuse.createIndex(options.keys, data);`

Error:
Cannot read property 'createIndex' of undefined

However when importing using the require `const Fuse = require('fuse.js')` there is no any errors.

I would like to know the reasons for these errors.

TS Config:

```
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es2017",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true
  },
  "exclude": ["node_modules", "dist"],
  "include": ["src"]
}
```

ENV Details:

Node.JS: v14.17.1
Typescript: v4.2.4
fuse.js: v6.5.3

Thank you very much.

**Comment by Vrajs16** [comment]
I am getting the same issue

**Comment by yaronveg** [comment]
Same here.
In my case I suspected a relation between how NodeNext requires imports with full extensions, i.e. `file.js`, and the package name: `fuse.js`.
I see now this issue appears in commonJS as well :(

`package.json`
```
{
  "name": "x",
  "version": "y",
  "description": "",
  "main": "index.js",
  "type": "module",
  "scripts": {
    "start": "nodemon src/app.ts",
    "build": "tsc"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "dotenv": "^16.3.1",
    "express": "^4.18.2",
    "fuse.js": "^6.6.2",
    "mongoose": "^7.3.4"
  },
  "devDependencies": {
    "@types/express": "^4.17.17",
    "@types/mongoose": "^5.11.97",
    "@types/node": "^20.4.2",
    "nodemon": "^3.0.1",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.6"
  }
}

``` 

`tsconfig.json`
```
{
    "compilerOptions": {
        "module": "NodeNext",
        "target": "ES5",
        "sourceMap": true,
        "outDir": "dist"
    },
    "include": ["src/**/*"],
    "ts-node": {
        "esm": true,
        "files": true
        // "experimentalSpecifierResolution": "node"
    }
}
```

**Comment by krisk** [comment, marked as answer]
Apologies for the very late reply.

This was a known issue with the CommonJS/ESM interop. The library now ships with proper dual-format support (`exports` map in `package.json` with separate `.mjs` and `.cjs` builds), so `import Fuse from 'fuse.js'` should work correctly in both ESM and CommonJS environments. If you're still hitting this, make sure you're on the latest version.
